File: libraries/observability-py/observability/context/trace_identity.py
from opentelemetry import trace

INVALID_TRACE_ID = 0
INVALID_SPAN_ID = 0


def current_trace_id_hex() -> str | None:
    span_context = trace.get_current_span().get_span_context()
    if span_context.trace_id == INVALID_TRACE_ID:
        return None

    return trace.format_trace_id(span_context.trace_id)


def current_span_id_hex() -> str | None:
    span_context = trace.get_current_span().get_span_context()
    if span_context.span_id == INVALID_SPAN_ID:
        return None

    return trace.format_span_id(span_context.span_id)

File: libraries/observability-go/tracing/trace_identity.go
package tracing

import (
	"context"

	"go.opentelemetry.io/otel/trace"
)

// CurrentTraceIDHex returns the active trace id, empty when nothing is sampled.
func CurrentTraceIDHex(ctx context.Context) string {
	spanContext := trace.SpanContextFromContext(ctx)
	if !spanContext.HasTraceID() {
		return ""
	}

	return spanContext.TraceID().String()
}

// CurrentSpanIDHex returns the active span id, empty when nothing is sampled.
func CurrentSpanIDHex(ctx context.Context) string {
	spanContext := trace.SpanContextFromContext(ctx)
	if !spanContext.HasSpanID() {
		return ""
	}

	return spanContext.SpanID().String()
}
